dame_flame/dame_algorithm.py

```diff
@@ -125,7 +125,14 @@ def algo1(df_all, treatment_column_name = "T", weight_array = [],
     all_covs.remove(outcome_column_name)
     df_unmatched = df_all.copy(deep=True) # This is df_h in the paper    
     all_covs_max_list = [max(df_all[x])+1 for x in all_covs] 
-                      
+    
+    #Ensuring that no entries are non-integers
+    for index, row in df_all.iterrows():
+        for i in all_covs:
+            assert (type(row[i])==int), 'Input has non-integer entries'
+
+
+                  
```

Comment:
Don't do this here. Find a good spot in the data_cleaning.py to put it, where it makes sense with the corresponding function names. Secondly, change all asserts to "raise exception"